debian/control

```diff
@@ -85,7 +85,7 @@ Build-Depends: automake,
                prometheus <pkg.ceph.check>,
                python3-all-dev,
                python3-cherrypy3,
-               python3-natsort,
+               python3-natsort <pkg.ceph.check>,
```

Comment:
No asyncssh too? I don't find that defined anywhere in this file 🤔

Reply:
> No asyncssh too? I don't find that defined anywhere in this file thinking

Yeah, I have to double check why this package has never been included on this file! .. maybe something is broken and nobody has never tested it!

Reply:
😲

Reply:
Ok, it looks like Debian packaging supports some magic tricks: the ${python3:Depends} is populated by https://manpages.debian.org/testing/dh-python/dh_python3.1.en.html from the requirements.txt files. It looks like rpmbuild supports [a similar mechanism](https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#Automatically-generated-dependencies). It would be great if we could get rid of these manually defined deps and use the requirements.txt as the source for packages too.

Reply:
If you check the `debian/rules` there the `dh_python3` is invoked for certain packages.

Reply:
i'd consider this change a cleanup. 

the change adding `python3-natsort` was introduced by #44895, which added this package as both build dependency and runtime dependency. i'd be disapprove that change BTW. as

1. `dh_python3` parses python3 scripts to find out used dependencies. so normally, we don't need to add python3 dependencies explicitly.
2. and we don't use `python3-natsort` for building Ceph. 


the better way to fix the runtime dependency is to add `ceph-mgr-modules-core` to the list of packages in `override_dh_python3` target in `debian/rules`.

Reply:
i'd consider this change a cleanup. 

the change adding `python3-natsort` was introduced by #44895, which added this package as both build dependency and runtime dependency. i'd be disapprove that change BTW. as

1. `dh_python3` parses python3 scripts to find out used dependencies. so normally, we don't need to add python3 dependencies explicitly.
2. and we don't use `python3-natsort` for building Ceph. 


the better way to fix the runtime dependency is to add `ceph-mgr-modules-core` to the list of packages in `override_dh_python3` target in `debian/rules`.

Reply:
> Ok, it looks like Debian packaging supports some magic tricks: the ${python3:Depends} is populated by https://manpages.debian.org/testing/dh-python/dh_python3.1.en.html from the requirements.txt files. It looks like 

not quite true, it parses the `.egg-info/requires.txt` not `requirements.txt`. the latter is not packaged, and should not be visible from `dh_python3`.

> rpmbuild supports [a similar mechanism](https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#Automatically-generated-dependencies). It would be great if we could get rid of these manually defined deps and use the requirements.txt as the source for packages too.

not sure if this applies to our mgr modules, which are not typical python packages in the sense that they are not packaged using `setuptool` or `pyproject.toml` . so we are on our own to improve the tooling to teach the packaging script the dependencies.

i think #46416 is the first step. with proper tooling we should be able to consolidate the dependencies scattered in both `debian/*.requires` and `.spec` into a single place -- `requirements.txt` located under each of the mgr module subdirectory.

Reply:
Thank you very much @epuertat and @tchaikov for helping with this :)

+1 for https://github.com/ceph/ceph/pull/46416 and for consolidating all the reqs in a single place. That would make devs life easier in addition to state clearly what are the deps for each mgr module.

PD: I'll revert changes on the `debian/control` since  https://github.com/ceph/ceph/pull/46416 will take care of them.

Reply:
@tchaikov please, let me know if any additional changes are needed to merge this PR.